File: DGT/ClientApp/src/app/models/crash-event/geocoding.ts
// Definition: The result of the automatic batch geocoding process by which a computer
// algorithm attempts to translate the textual description of a crash location into an
// unambiguous point on the Florida All Roads Base Map (ARBM) including explicit
// relationships to the nearest roadway segment, nearest intersection, and offset
// distance and direction from the nearest intersection.
import { Editor } from 'app/models/users/editor';
import {MapPoint} from "../geolocation/map-point";

export interface Geocoding {
  hsmvReportNumber: number;
  mapPoints: MapPoint[];
  createDate?: Date | undefined;
  lastUpdateDate?: Date | undefined;
  etlGeoLocationStatus: string;
  relationShipToNetwork?: string;
  centerlineX?: number;
  centerlineY?: number;
  matchStatusCode?: string;
  matchResultCode?: number;
  failReasonCode?: string;
  crashSegmentId?: string;
  nearestIntersectionId?: string;
  nearestIntersectionOffsetFeet?: number;
  nearestIntersectionOffsetDirection?: string;
  mapped?: boolean;
  onNetwork?: boolean;
  author?: string;
}

export interface EditedGeocoding extends Geocoding {
  poolId?: number;
  userId?: number;
  fdotGeocodeStatus?: string;
  comments?: string;
  createUser?: Editor;
  lastUpdateUser?: Editor;
}

File: DGT/Models/Geocoding.cs
using System;
using System.Collections.Generic;

namespace DGT.Models
{
    public class Geocoding
    {
        private int _hsmvReportNumber;
        private List<MapPoint> _mapPoints;
        private DateTime _createDate;
        private DateTime _lastUpdateDate;
        private string _etlGeoLocationStatus;
        private string _relationShipToNetwork;
        private double _centerlineX;
        private double _centerlineY;
        private string _matchStatusCode;
        private int _matchResultCode;
        private int  _failReasonCode;
        private string _crashSegmentId;
        private string _nearestIntersectionId;

        public class MapPoint
        {
            private double _x;
            private double _y;

            public double X
            {
                get => _x;
                set => _x = value;
            }

            public double Y
            {
                get => _y;
                set => _y = value;
            }
        }
        
        public List<MapPoint> MapPoints
        {
            get => _mapPoints;
            set => _mapPoints = value;
        }

        public int HsmvReportNumber
        {
            get => _hsmvReportNumber;
            set => _hsmvReportNumber = value;
        }

        public DateTime CreateDate
        {
            get => _createDate;
            set => _createDate = value;
        }

        public DateTime LastUpdateDate
        {
            get => _lastUpdateDate;
            set => _lastUpdateDate = value;
        }

        public string EtlGeoLocationStatus
        {
            get => _etlGeoLocationStatus;
            set => _etlGeoLocationStatus = value;
        }

        public string RelationShipToNetwork
        {
            get => _relationShipToNetwork;
            set => _relationShipToNetwork = value;
        }

        public double CenterlineX
        {
            get => _centerlineX;
            set => _centerlineX = value;
        }

        public double CenterlineY
        {
            get => _centerlineY;
            set => _centerlineY = value;
        }

        public string MatchStatusCode
        {
            get => _matchStatusCode;
            set => _matchStatusCode = value;
        }

        public int MatchResultCode
        {
            get => _matchResultCode;
            set => _matchResultCode = value;
        }

        public int FailReasonCode
        {
            get => _failReasonCode;
            set => _failReasonCode = value;
        }

        public string CrashSegmentId
        {
            get => _crashSegmentId;
            set => _crashSegmentId = value;
        }

        public string NearestIntersectionId
        {
            get => _nearestIntersectionId;
            set => _nearestIntersectionId = value;
        }
    }
}
